### PR TITLE
Merge view-change event with synthetic decisioning.propositionDisplay notification event 

### PR DESCRIPTION
**Pull request by ninaceban**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

**Review by jonsnyder — COMMENTED** [review]
This is a great new feature, but I do feel like the code is getting complex. One thing that could simplify the code is to separate the concerns of executing the decisions and recording the display/view change notifications.

So change the executeDecisions function to always return the decisionsMeta. Then create a recordNotification function to either merge the decisionsMeta onto the event, or generate its own notifications. recordNotification might look something like this:

```
const createRecordNotifications = ({ mergeDecisionsMeta, collect, sendEmptyViewNotification }) => ({ viewName, event, decisionsMeta }) => {
  if (isNonEmptyArray(decisionsMeta)) {
    if (event) {
      // include the display notification on the event
      mergeDecisionsMeta(event, decisionsMeta);
    } else {
      // send a separate display notification
      collect({ decisionsMeta })
    }
  } else {
    // send an empty view notification
    sendEmptyViewNotification(viewName);
  }
}
```

File: src/components/Personalization/createExecuteDecisions.js [hunk 1]
```diff
@@ -49,21 +49,32 @@ const processMetas = (collect, logger, actionResults) => {
...
+  return (decisions, returnDecisionsMeta = false) => {
     const actionResultsPromises = decisions.map(decision => {
       const actions = buildActions(decision);
 
       return executeActions(actions, modules, logger);
     });
     return Promise.all(actionResultsPromises)
-      .then(results => processMetas(collect, logger, results))
+      .then(results =>
+        processMetas(collect, logger, results, returnDecisionsMeta)
```
Comment (5amfung) [inline src/components/Personalization/createExecuteDecisions.js:64]: `processMetas()` at line 27 doesn't seem to have the 4th parameter.

File: src/components/Personalization/createCollect.js [hunk 2]
```diff
@@ -10,13 +10,24 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 import { DISPLAY } from "./constants/eventType";
+import { isNonEmptyArray } from "../../utils";
 
 export default ({ eventManager, mergeDecisionsMeta }) => {
   // Called when a decision is auto-rendered for the __view__ scope (non-SPA view).
```
Comment (jonsnyder) [inline src/components/Personalization/createCollect.js:16]: Can you update this comment?

File: src/components/Personalization/createExecuteDecisions.js [hunk 3]
```diff
@@ -52,28 +52,17 @@ const processMetas = (collect, logger, actionResults) => {
...
-        }
-
-        if (isNonEmptyArray(finalMetas)) {
-          // collect here can either be the function from createCollect or createViewCollect.
-          collect({ decisionsMeta: finalMetas });
-        }
-
-        return {};
+      .then(results => processMetas(logger, results))
+      .then(decisionsMeta => {
```
Comment (jonsnyder) [inline src/components/Personalization/createExecuteDecisions.js:64]: Is this necessary?

**Review by jonsnyder — APPROVED** [review]
This looks great!

File: src/components/Personalization/createExecuteDecisions.js (hunk 1 above)
Comment (jfkhoury) [reply src/components/Personalization/createExecuteDecisions.js]: @ninaceban Does this comment still apply?

Comment (ninaceban) [reply src/components/Personalization/createExecuteDecisions.js]: nope, this file was completely refactored. Thank you

File: src/components/Personalization/createExecuteDecisions.js (hunk 3 above)
Comment (ninaceban) [reply src/components/Personalization/createExecuteDecisions.js:64]: nope, this should have been removed.

File: src/components/Personalization/createAutoRenderingHandler.js [hunk 4]
```diff
@@ -11,28 +11,59 @@ governing permissions and limitations under the License.
...
+
+        if (viewName) {
+          executeDecisions(propositions.viewPropositions).then(
+            decisionsMeta => {
+              collect({ decisionsMeta, viewName });
+            }
+          );
+        }
+
         showContainers();
```
Comment (jonsnyder) [inline src/components/Personalization/createAutoRenderingHandler.js:54]: Why are we showing containers before the decisions are executed?

**Review by jonsnyder — APPROVED** [review]

**Pull request merged**